### PR TITLE
Custom error for homebrew table not found

### PR DESCRIPTION
**Pull request by Dharma-09**
Which issue(s) this PR fixes:
#1864 

Description:
In the given solution, an error is returned every time an issue occurs during the `validatedCommand` call, regardless of the cause. For instance, if the binary file does not exist, should we use a specific error to handle this scenario?

### CODE REVIEW

File: ee/allowedcmd/cmd_linux.go
Comment (RebeccaMahany) [inline ee/allowedcmd/cmd_linux.go:16]: I am wondering if we would instead want to move this to `cmd.go`, rename it something more generic like `ErrCommandNotFound` and return it from the `validatedCommand` function -- this new error is very useful here, and I imagine we might want to do similar in the future when other commands aren't found. 

(This will also have the bonus side effect of fixing the build errors, since we run `Brew` on both macOS and Linux so this new error needs to be available to darwin builds as well.)

Comment (Dharma-09) [reply ee/allowedcmd/cmd_linux.go:16]: That's great idea, I'll update the PR with the changes.

File: ee/allowedcmd/cmd_linux.go
Comment (RebeccaMahany) [inline ee/allowedcmd/cmd_linux.go:19]: I think this should still be `nil, err`, so we can return the wrapped `ErrCommandNotFound` to `upgradeable.go` to check

File: ee/allowedcmd/cmd.go
Comment (RebeccaMahany) [inline ee/allowedcmd/cmd.go:37]: Tiny nitpick -- I think the spacing is off here --

```suggestion
		return nil, fmt.Errorf("%w: %s", ErrCommandNotFound, knownPath)
```

File: ee/allowedcmd/cmd.go
Comment (RebeccaMahany) [inline ee/allowedcmd/cmd.go:45]: Another nitpick -- I think the error message will read more clearly like this:

```suggestion
	return nil, fmt.Errorf("%w: not found at %s and could not be located elsewhere", ErrCommandNotFound, knownPath)
```

**Review by RebeccaMahany — COMMENTED** [review]
Just a couple small comments, looks good to me overall! Will hold off on approval till @directionless or another team member gets a chance to look as well 🙂

File: ee/tables/homebrew/upgradeable.go
Comment (RebeccaMahany) [inline ee/tables/homebrew/upgradeable.go:51]: I think you updated the wrong one -- this should still be `return nil, nil` as you had it before 🙂

**Review by directionless — APPROVED** [review]
I think Becca covered all my comments. Thank you so much for you contribution!

File: ee/allowedcmd/cmd.go
Comment (RebeccaMahany) [inline ee/allowedcmd/cmd.go:15]: It looks like the lint job is complaining because this import is not added alphabetically -- same goes for other file as well

**Review by RebeccaMahany — COMMENTED** [review]
LGTM once the lint passes!

File: ee/tables/homebrew/upgradeable.go
Comment (RebeccaMahany) [inline ee/tables/homebrew/upgradeable.go:47]: Apologies, missed this previously -- looks like the lint job is complaining about this line as well. (If you want to validate for yourself before CI, you can [install golangci-lint](https://golangci-lint.run/welcome/install/#local-installation) and then run `golangci-lint run`.)

**Review by RebeccaMahany — APPROVED** [review]
🔥

**Pull request merged**
